node/p2p_block_sync.md

```diff
@@ -0,0 +1,81 @@
+# P2P Block Sync
+
+## Abstract
+
+P2P Block Sync enables rollkit full nodes including aggregators to gossip blocks amongst  
+themselves and sync with the rollup chain faster than they can sync using the DA layer.
+
+```mermaid
+sequenceDiagram
+    title P2P Block Sync
+
+    participant User
+    participant Block Producer
+    participant Full Node 1
+    participant Full Node 2
+    participant DA Layer
+
+    User->>Block Producer: Send Tx
+    Block Producer->>Block Producer: Generate Block
+    Block Producer->>DA Layer: Publish Block
+
+    Block Producer->>Full Node 1: Gossip Block
+    Block Producer->>Full Node 2: Gossip Block
+    Full Node 1->>Full Node 1: Verify Block
+    Full Node 1->>Full Node 2: Gossip Block
+    Full Node 1->>Full Node 1: Mark Block Soft-Confirmed
+
+    Full Node 2->>Full Node 2: Verify Block
+    Full Node 2->>Full Node 2: Mark Block Soft-Confirmed
+
+    DA Layer->>Full Node 1: Retrieve Block
+    Full Node 1->>Full Node 1: Mark Block Hard-Confirmed
+
+    DA Layer->>Full Node 2: Retrieve Block
+    Full Node 2->>Full Node 2: Mark Block Hard-Confirmed
+```
+
+## Protocol/Component Description
+
+P2P Block Sync consists of the following components:
+
+* block exchange service: responsible for gossiping blocks over P2P
+* block publication to P2P network
+* block retrieval from P2P network
+
+### Block Exchange Service
+
+The block exchange service uses the ([`go-header` service](https://github.com/celestiaorg/go-header)) to gossip blocks over P2P.
```

Comment:
it is good to deduplicate by referencing the go-header repo, but we should maybe include the API/Methods used here, or explain the functionality expected.  Otherwise we aren't able to audit that we are using go-header properly.

Reply:
Added. Makes it clear that we have full node initializes block exchange service -> passes the block store inside it to initialize block manager which then publishes blocks to P2P network through a channel -> those blocks get gossiped over P2P -> blocks arrive on block stores inside the other full nodes -> these are retrieved in their block managers.